pymepixviewer/panels/timepixsetupplotspanel.py

```diff
@@ -0,0 +1,205 @@
+
+        return result
+
+    def __setup_2d_hist_ui(self, name, position, label_left, label_bottom):
+        plt_2d_hist = pg.ImageView(view=pg.PlotItem())
+        plt_2d_hist.setObjectName(name)
+        self.gridLayout.addWidget(plt_2d_hist, position[0], position[1], 1, 1)
+        
+        plt_2d_hist.setPredefinedGradient("thermal")
+        plt_2d_hist.getView().setLabel('bottom', text=label_left)
+        plt_2d_hist.getView().setLabel('left', text=label_bottom)
+        plt_2d_hist.getView().invertY(False)
+        plt_2d_hist.getView().setAspectLocked(False)
+
+        return plt_2d_hist
+
+    def onNumberPacketsChanged(self, number_packets):
+        if number_packets == 0:
+            number_packets = None
+            self.pushButtonReset.setEnabled(True)
+        else:
+            self.pushButtonReset.setEnabled(False)
+        self.__init_buffers(number_packets)
+
+    def __init_buffers(self, number_packets=None):
+        self._event_data_tot_histogram.init_buffer(number_packets)
+        self.__events_2d_hist_buffer = deque(maxlen=number_packets)
+
+        self._centroided_data_mean_tot_histogram.init_buffer(number_packets)
+        self._centroided_data_max_tot_histogram.init_buffer(number_packets)
+        self.__centroids_2d_hist_mean_buffer = deque(maxlen=number_packets)
+        self.__centroids_2d_hist_max_buffer = deque(maxlen=number_packets)
+        self._centroided_data_cluster_size_histogram.init_buffer(number_packets)
+
+    def __reset_buffers(self):
+        self.__reset_tof_buffers()
+        self.__reset_tot_buffers()
+
+    def __reset_tof_buffers(self):
+        if self.__events_2d_hist_buffer is not None:
+            self.__events_2d_hist_buffer.clear()
+        if self.__centroids_2d_hist_mean_buffer is not None:
+            self.__centroids_2d_hist_mean_buffer.clear()
+        if self.__centroids_2d_hist_max_buffer is not None:
+            self.__centroids_2d_hist_max_buffer.clear()
+
+    def __reset_tot_buffers(self):
+        self._event_data_tot_histogram.clear_buffer()
+        if self.__events_2d_hist_buffer is not None:
+            self.__events_2d_hist_buffer.clear()
+
+        self._centroided_data_mean_tot_histogram.clear_buffer()
+        self._centroided_data_max_tot_histogram.clear_buffer()
+        if self.__centroids_2d_hist_mean_buffer is not None:
+            self.__centroids_2d_hist_mean_buffer.clear()
+        if self.__centroids_2d_hist_max_buffer is not None:
+            self.__centroids_2d_hist_max_buffer.clear()
+
+    def __snapshot(self):
+        self._event_data_tot_histogram.snapshot(self.__tot_bins)
+        self._centroided_data_max_tot_histogram.snapshot(self.__tot_bins)
+        self._centroided_data_mean_tot_histogram.snapshot(self.__tot_bins)
+        self._centroided_data_cluster_size_histogram.snapshot(CLUSTER_SIZE_BINS)
+
```

Comment:
Would be cool, if Python/Qt had a decorator here to connect the functions to the signals.